Content/src/Server/ServerGiraffe.fs

```diff
@@ -1,34 +1,55 @@
-open System
-open System.IO
-open System.Threading.Tasks
-
+#if (remoting)
+open Fable.Remoting.Server
+open Fable.Remoting.Giraffe
+#endif
+#if (deploy == "azure")
+open FSharp.Azure.StorageTypeProvider
+#endif
+open FSharp.Control.Tasks.V2
+open Giraffe
+#if (deploy == "azure")
+open global.Shared
+#endif
 open Microsoft.AspNetCore
 open Microsoft.AspNetCore.Builder
 open Microsoft.AspNetCore.Hosting
 open Microsoft.Extensions.DependencyInjection
-
-open FSharp.Control.Tasks.V2
-open Giraffe
+#if (deploy != "azure")
 open Shared
-
-#if (remoting)
-open Fable.Remoting.Server
-open Fable.Remoting.Giraffe
 #endif
+open System
+open System.IO
 #if (deploy == "azure")
-open Microsoft.WindowsAzure.Storage
+open Thoth.Json.Net
 #endif
 
-//#if (deploy == "azure")
+#if (deploy == "azure")
 let tryGetEnv = System.Environment.GetEnvironmentVariable >> function null | "" -> None | x -> Some x
 let publicPath = tryGetEnv "public_path" |> Option.defaultValue "../Client/public" |> Path.GetFullPath
-let storageAccount = tryGetEnv "STORAGE_CONNECTIONSTRING" |> Option.defaultValue "UseDevelopmentStorage=true" |> CloudStorageAccount.Parse
-//#else
+let runtimeAzure = tryGetEnv "STORAGE_CONNECTIONSTRING" |> Option.defaultValue "UseDevelopmentStorage=true"
+
+/// A handle to an Azure storage account without a connection string. Schema is provided by the
+/// azure-schema.json file. You can remove the blobSchema key/value and replace with a full Azure
+/// connection string; schema will be inferred from the storage account contents directly.
+type Azure = AzureTypeProvider<blobSchema="azure-schema.json">
+let safeData = Azure.Containers.safedata
+#else
 let publicPath = Path.GetFullPath "../Client/public"
-//#endif
+#endif
 let port = 8085us
 
-let getInitCounter () : Task<Counter> = task { return { Value = 42 } }
+#if (deploy == "azure")
+// Initialise the Azure storage account with a container and the counter state.
+do
+    safeData.AsCloudBlobContainer(runtimeAzure).CreateIfNotExistsAsync().Wait()
```

Comment:
Should we be encouraging people to create containers in code still or should we instead start creating containers within the ARM template? https://samcogan.com/create-azure-storage-containers-with-arm-templates/

Reply:
That's a nice new addition to ARM. However, how will this work from a local development perspective (think about the out-of-the-box experience)